python/GangaLHCb/Lib/Applications/GaudiExec.py

```diff
@@ -204,11 +204,11 @@ def prepare(self, force=False):
         # file is unspecified, has a space or is a relative path
         self.configure(self)
         logger.info('Preparing %s application.' % getName(self))
-        self.is_prepared = ShareDir()
-        logger.info('Created shared directory: %s' % (self.is_prepared.name))
 
         this_build_target = self.buildGangaTarget()
 
+        self.is_prepared = ShareDir()
+        logger.info('Created shared directory: %s' % (self.is_prepared.name))
```

Comment:
@rob-c Looking at the code I don't think the buildTarget requires the sharedDir to exist yet. The stuff in the try though does.

I agree that is_prepared should be removed in the exception though.